README.md

```diff
@@ -0,0 +1,376 @@
+
+## Usage
+
+### Publishing messages
+
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+
+	redisSyncFanoutQueue "github.com/zavitax/redis-sync-fanout-queue-go"
+)
+
+var testMessageContent = "test message content"
+var testRoomId = "GO-ROOM-TEST"
+
+var redisOptions = &redis.Options{
+	Addr:     "127.0.0.1:6379",
+	Password: "",
+	DB:       0,
+}
+
+func createApiOptions() *redisSyncFanoutQueue.ApiOptions {
+	result := &redisSyncFanoutQueue.ApiOptions{
+		RedisOptions:   redisOptions,
+		ClientTimeout:  time.Second * 15,
+		RedisKeyPrefix: fmt.Sprintf("{test-redis-sync-fanout-queue}::%v", "test"),
+	}
+
+	return result
+}
+
+func createApiClient(options *redisSyncFanoutQueue.ApiOptions) (redisSyncFanoutQueue.RedisQueueApiClient, error) {
+	return redisSyncFanoutQueue.NewApiClient(context.TODO(), options)
+}
+
+func pub() {
+	doneC := make(chan os.Signal, 1)
+	signal.Notify(doneC)
+
+	client, _ := createApiClient(createApiOptions())
+
+	i := 0
+	ticker := time.NewTicker(time.Second)
+	done := false
+	for !done {
+		select {
+		case <-doneC:
+			done = true
+		case <-ticker.C:
+			i++
+
+			client.Send(context.TODO(), "MSG_PRODUCER_ID", testRoomId, testMessageContent, 1)
+
+			fmt.Printf("Send: %v\r", i)
+		}
+	}
+	ticker.Stop()
+
+	client.Close()
+}
+
+func main() {
+	pub()
+}
+```
+
+### Subscribing and maintaining a client connection
+
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+
+	redisSyncFanoutQueue "github.com/zavitax/redis-sync-fanout-queue-go"
+)
+
+var testMessageContent = "test message content"
+var testRoomId = "GO-ROOM-TEST"
+
+var redisOptions = &redis.Options{
+	Addr:     "127.0.0.1:6379",
+	Password: "",
+	DB:       0,
+}
+
+func createApiOptions() *redisSyncFanoutQueue.ApiOptions {
+	result := &redisSyncFanoutQueue.ApiOptions{
+		RedisOptions:   redisOptions,
+		ClientTimeout:  time.Second * 15,
+		RedisKeyPrefix: fmt.Sprintf("{test-redis-sync-fanout-queue}::%v", "test"),
+	}
+
+	return result
+}
+
+func createApiClient(options *redisSyncFanoutQueue.ApiOptions) (redisSyncFanoutQueue.RedisQueueApiClient, error) {
+	return redisSyncFanoutQueue.NewApiClient(context.TODO(), options)
+}
+
+func sub(clientsCount int) {
+	doneC := make(chan os.Signal, 1)
+	signal.Notify(doneC)
+
+	client, _ := createApiClient(createApiOptions())
+
+	clientIds := []string{}
+	for i := 0; i < clientsCount; i++ {
+		clientId, _ := client.CreateClientID(context.Background())
+		client.Subscribe(context.TODO(), clientId, testRoomId)
+		clientIds = append(clientIds, clientId)
+	}
+
+	ticker := time.NewTicker(time.Second)
+	done := false
+	for !done {
+		select {
+		case <-doneC:
+			done = true
+		case <-ticker.C:
+			for _, clientId := range clientIds {
+				// Must call periodically for each Client ID & Room ID comibnation to keep
+				// Client ID alive.
+				client.Ping(context.Background(), clientId)
```

Comment:
There is clear separation between client requests to the library (API) and message delivery (worker).
The other way around would be to have a `worker` process for every client ID in the system, which is clearly not scalable.
Clients talk to the API to subscribe for message delivery, and to inform the library they are still alive.
Actual messages are delivered to clients over an unspecified mechanism by async workers.
Under those constraints, you can't rely on the library to maintain keep-alive messages, since it simply doesn't maintain an open connection with the client.